### ISSUE TITLE
Misleading behavior if user provides a single command-line arg

### ISSUE THREAD

**Issue by sethladd** [dart-lang/dartdoc#591]
Consider this:

```
~/Code/dartdoc[new-layout*] $ dart bin/dartdoc.dart test_package
Generating documentation for 'dartdoc' into /Users/sethladd/Code/dartdoc/doc/api/.
```

The intention was "generate the docs for the code in the `test_package`. Regardless if that was a reason expectation or user error :) notice how dartdoc started generating docs for the current directory.

What do you think about throwing an error if a user passes a command-line arg like this? The tool doesn't currently do anything with that, and the behavior might be misleading.

Or... we could use the command-line arg as the location of the input.

**Comment by sethladd** [comment]
@keertip @devoncarew any thoughts?

**Comment by devoncarew** [comment]
+1 to either throwing if args.rest is not empty, or using that as the cwd for generating docs. the first option is simpler; the 2nd could be a later enhancement.

**Comment by sethladd** [comment]
Let's go with throwing if args.rest is empty for now.

**Comment by sethladd** [comment]
OK, we print an error now:

```
~/Code/dartdoc[throw-on-invalid-command-line-args] $ dart bin/dartdoc.dart foo
Error: detected unknown command-line argument(s)
Usage: dartdoc [OPTIONS]

-h, --help            Show command help.
    --version         Display the version for dartdoc.
    --dart-sdk        Location of the Dart SDK. Use if SDK isn't automatically located.
    --sdk-docs        Generate ONLY the docs for the Dart SDK.
    --sdk-readme      Path to the SDK description file. Use if generating Dart SDK docs.
    --input           Path to source directory
                      (defaults to "/Users/sethladd/Code/dartdoc")

    --output          Path to output directory.
                      (defaults to "doc/api")

    --header          path to file containing HTML text, inserted into the header of every page.
    --footer          path to file containing HTML text, inserted into the footer of every page.
    --package-root    The path to the package root.
    --exclude         Comma-separated list of library names to ignore.
    --hosted-url      URL where the docs will be hosted (used to generate the sitemap).
```

**Comment by sethladd** [comment]
Fixed with d7397ab
